### PR TITLE
adding more detailed logging in the event that a test rule fails to apply

### PR DESCRIPTION
**Pull request by burnMyDread**
I have some tests running with the withPlugins test rule and a few others. I would really like a way to tell which of my test rules is failing when the tests fail to apply all the test rules. To that end I am submitting this PR to report which test rule failed and the throwable's cause. I didn't see a contributors guide for this, so if I missed it please let me know. @reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/test/acceptance/junit/JenkinsAcceptanceTestRule.java [hunk 1]
```diff
@@ -110,7 +110,12 @@ public int compare(Integer o1, Integer o2) {
 
                 for (Set<TestRule> rulesGroup: rules.values()) {
                     for (TestRule rule: rulesGroup) {
-                        body = rule.apply(body, description);
+                        try {
+                            body = rule.apply(body, description);
+                        } catch (Exception e) { //This is to note which rule is failing when a rule does fail as the evaluate method hinds this information
+                            System.out.println("Test Rule " + rule.getClass() + " failed to apply. Failed with: " + e.getCause());
+                            throw e;
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/test/acceptance/junit/JenkinsAcceptanceTestRule.java]: I would be better to wrap to another exception indicating the rule class in message as this way the stack trace and the rule name are both reported but by separate mechanisms.

Comment (burnMyDread) [reply src/main/java/org/jenkinsci/test/acceptance/junit/JenkinsAcceptanceTestRule.java]: That was my original plan but I didn't want to change the exception type as that was what is sent to the controller in the diagnose method. I can change it to a warped exception.

**Comment by burnMyDread** [comment]
@olivergondza I move to a service exception model, please take a look.

**Comment by olivergondza** [comment]
Merging amended with https://github.com/jenkinsci/acceptance-test-harness/commit/8c74b90b1b03c8aa10e3a5234e189a91cd090c49. Thanks

**Pull request merged**